airbus_harvester/__main__.py

```diff
@@ -146,65 +159,38 @@ def harvest(workspace_name: str, catalog: str, s3_bucket: str):
         last_run_hash = latest_harvested.get(collection_key)
         previous_hash = last_run_hash if last_run_hash else previously_harvested.get(collection_key)
 
-        harvested_keys, latest_harvested[collection_key] = compare_to_previous_version(
-            collection_key, collection_data, previous_hash, harvested_keys, s3_bucket, s3_client
-        )
+        file_hash = get_file_hash(json.dumps(collection_key))
+        if not previous_hash or previous_hash != file_hash:
+            # Data was not harvested previously
+            logging.info(f"Added: {collection_key}")
+            harvested_data[collection_key] = collection_data
+            latest_harvested[collection_key] = file_hash
 
         latest_harvested["summary"] = catalogue_data_summary
 
-        # Don't send empty or nearly empty pulsar messages
-        if (
-            len(harvested_keys["added_keys"])
-            + len(harvested_keys["updated_keys"])
-            + len(deleted_keys)
-            > minimum_message_entries
-        ):
-            # Record harvested hash data in S3
-            upload_file_s3(json.dumps(latest_harvested), s3_bucket, metadata_s3_key, s3_client)
-
-            output_data = {
-                "id": f"{workspace_name}/{config['collection_name']}_{url_count}",
-                "workspace": workspace_name,
-                "bucket_name": s3_bucket,
-                "added_keys": list(harvested_keys["added_keys"]),
-                "updated_keys": list(harvested_keys["updated_keys"]),
-                "deleted_keys": deleted_keys,
-                "source": "airbus",
-                "target": "/",
-            }
-
-            send_pulsar_message(output_data, pulsar_client)
-
-            harvested_keys = {"added_keys": set(), "updated_keys": set()}
+        if len(harvested_data.keys()) >= minimum_message_entries:
 
-    upload_file_s3(json.dumps(latest_harvested), s3_bucket, metadata_s3_key, s3_client)
+            # Send message for altered keys
+            msg = {
```

Comment:
Does msg get used before the end? Should there be something here to send the pulsar message?